### PR TITLE
Add support for Checkout card and customer details.

### PR DESCRIPTION
**Pull request by mizinsky**
This PR will add support for updating card and customer details.

### CODE REVIEW

**Comment by mizinsky** [comment]
Screenshot 2021-08-19 at 19 49 28

File: lib/active_merchant/billing/gateways/checkout_v2.rb [hunk 1]
```diff
@@ -243,13 +275,7 @@ def add_3ds(post, options)
 
       def commit(action, post, authorization = nil)
         begin
-          raw_response = if action == :verify_payment
-            ssl_get("#{base_url}/payments/#{post}", headers(action))
-          elsif action == :delete_card
-            ssl_request(:delete, url(post, action, authorization), nil, headers(action))
-          else
-            ssl_post(url(post, action, authorization), post.to_json, headers(action))
-          end
```
Comment (dbackowski) [inline lib/active_merchant/billing/gateways/checkout_v2.rb:252]: 👍

File: lib/active_merchant/billing/gateways/checkout_v2.rb [hunk 2]
```diff
@@ -177,23 +201,31 @@ def add_payment_method(post, payment_method, options)
...
+          billing_address = { billing_address: {} }
           billing_address[:billing_address][:address_line1] = address[:address1] unless address[:address1].blank?
           billing_address[:billing_address][:address_line2] = address[:address2] unless address[:address2].blank?
           billing_address[:billing_address][:city] = address[:city] unless address[:city].blank?
           billing_address[:billing_address][:state] = address[:state] unless address[:state].blank?
           billing_address[:billing_address][:country] = address[:country] unless address[:country].blank?
           billing_address[:billing_address][:zip] = address[:zip] unless address[:zip].blank?
-          post[:source] ? post[:source].merge!(billing_address) : post.merge!(billing_address)
+          if billing_address_key
+            (post[billing_address_key] ||= {}).merge!(billing_address)
```
Comment (dbackowski) [inline lib/active_merchant/billing/gateways/checkout_v2.rb:225]: I think that this version is much more readable
```ruby
post[billing_address_key] ||= {}
post[billing_address_key].merge!(billing_address)
```
What do you think?

Comment (mizinsky) [reply lib/active_merchant/billing/gateways/checkout_v2.rb]: I just changed it.

**Review by dbackowski — APPROVED** [review]
👍 🚀

**Pull request merged**